packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -117,7 +117,7 @@ const _versionGroupName = 'version';
 // Kotlin DSL - ("com.android.tools.build.gradle:{{agpVersion}}")
 // ?<version> is used to name the version group which helps with extraction.
```

Comment:
Consider documenting the *entire* thing being matched, including why the leading `[^/]*\s*` is there.
(You can't debug a RegExp if you don't know, without looking at the RegExp, what it should match.)